### PR TITLE
use MUI Table in job list view

### PR DESCRIPTION
**Pull request by dlqqq**
This set of changes integrates the MUI Table component into the job list view. Pagination, sorting, and loading states are correctly handled.

https://user-images.githubusercontent.com/44106031/192384710-f15af7d5-07da-430c-8c23-a355fc0b573e.mov

What this PR does NOT include:

- does not open the job details view upon clicking on job name (waiting for @andrii-i 's detail view to be merged)
- does not remove `job-detail.tsx` despite being unused, since the contents of this file may be replaced by @andrii-i

### CODE REVIEW

**Comment by jweill-aws** [comment]
Gut reaction from watching the video: sorting (especially by start time) changes the column widths, which I find jarring. Would it be possible to fix the column widths to avoid this?

Also, there's still an open question about whether the Action buttons should be removed (we can wait until the Job Details is in to settle this)

File: src/mainviews/list-jobs.tsx [hunk 1]
```diff
@@ -52,17 +55,33 @@ export function NotebookJobsListBody(
...
+  const [page, setPage] = useState<number>(0);
+  const [maxPage, setMaxPage] = useState<number>(0);
+  const [loading, setLoading] = useState<boolean>(false);
+
+  const deleteRow = useCallback((id: Scheduler.IDescribeJob['job_id']) => {
+    setDeletedRows(deletedRows => new Set([...deletedRows, id]));
+  }, []);
+
+  const fetchInitialRows = async () => {
+    // reset pagiantion state
```
Comment (jweill-aws) [inline src/mainviews/list-jobs.tsx:70]:
```suggestion
    // reset pagination state
```

Fix typo

**Comment by dlqqq** [comment]
Rebased and resolved conflicts w/ `main`.

File: style/base.css [hunk 2]
```diff
@@ -113,15 +27,7 @@ a.jp-notebook-job-name {
...
 
-.jp-notebook-jobs-list-container {
-  margin-left: 12px;
-  margin-right: 12px;
-}
-
-ul.jp-notebook-job-output-formats-options {
-  list-style: none;
-  margin: 0 0 0 -40px;
-}
```
Comment (jweill-aws) [inline style/base.css:124]: Love all this CSS class cleanup ✨

**Comment by jweill-aws** [comment]
The table is sized at `0.75rem`, which seems a bit smaller than the previous version, and a little smaller than would be comfortable. What would it look like at an unmodified font size?

**Review by jweill-aws — APPROVED** [review]

File: src/mainviews/list-jobs.tsx [hunk 3]
```diff
@@ -148,38 +191,77 @@ export function NotebookJobsListBody(
...
+              {rows.slice(page * PAGE_SIZE, (page + 1) * PAGE_SIZE)}
+            </TableBody>
+          </Table>
+          <TablePagination
+            component="div"
+            sx={{
+              position: 'sticky',
+              bottom: 0,
+              backgroundColor: 'white',
+              borderTop: '1px solid #e0e0e0'
```
Comment (3coins) [inline src/mainviews/list-jobs.tsx:252]: These will not work with themes. We should find a way to use the JL theme vars here.

File: src/mainviews/list-jobs.tsx [hunk 4]
```diff
@@ -148,38 +191,77 @@ export function NotebookJobsListBody(
...
+    .filter(job => !deletedRows.has(job.job_id))
+    .map(job =>
+      buildTableRow(
+        job,
+        props.app,
+        props.showCreateJob,
+        deleteRow,
+        translateStatus
+      )
+    );
```
Comment (3coins) [inline src/mainviews/list-jobs.tsx:204]: Building all JSX elements here, but only using a slice of data at the bottom. Would there be a way to do the slicing of data first and only use JSX rows for the sliced data?

**Review by 3coins — COMMENTED** [review]
minor: Also noticed that the icons for sort are mis-aligned.

File: src/components/job-row.tsx [hunk 5]
```diff
@@ -1,16 +1,20 @@
...
 import { closeIcon, stopIcon } from '@jupyterlab/ui-components';
 
+import Stack from '@mui/material/Stack';
+import TableRow from '@mui/material/TableRow';
+import TableCell from '@mui/material/TableCell';
+
 import { Scheduler } from '../handler';
 import { useTranslator } from '../hooks';
 import { IJobParameter, ICreateJobModel } from '../model';
+import { CommandIDs } from '..';
```
[the remark this answers is not in the document]
Comment (dlqqq) [reply src/components/job-row.tsx:15]: Just a shorthand for `../index` in ESM import syntax. If a source file is named `index`, only the path to its parent directory is needed.

File: src/mainviews/list-jobs.tsx (hunk 3 above)
Comment (dlqqq) [reply src/mainviews/list-jobs.tsx]: Thank you. Very observant.

File: src/mainviews/list-jobs.tsx (hunk 4 above)
Comment (dlqqq) [reply src/mainviews/list-jobs.tsx]: Again, great comment. Fixed.

**Pull request merged**